server/coordinator/scheduler/nodepicker/hash/consistent_uniform.go

```diff
@@ -195,20 +224,52 @@ func (c *ConsistentUniformHash) distributePartitions() {
 	}
 }
 
-func (c *ConsistentUniformHash) MinLoad() float64 {
-	return c.minLoad
+func (c *ConsistentUniformHash) MinLoad() uint {
+	return uint(c.minLoad)
+}
+
+func (c *ConsistentUniformHash) MaxLoad() uint {
+	return uint(c.maxLoad)
 }
 
-func (c *ConsistentUniformHash) MaxLoad() float64 {
-	return c.maxLoad
+// LoadDistribution exposes load distribution of members.
+func (c *ConsistentUniformHash) LoadDistribution() map[string]uint {
+	loads := make(map[string]uint, len(c.memPartitions))
+	for member, partitions := range c.memPartitions {
+		loads[member] = uint(len(partitions))
+	}
+	return loads
+}
+
+// GetPartitionOwner returns the owner of the given partition.
+func (c *ConsistentUniformHash) GetPartitionOwner(partID int) Member {
+	virtualNodeIdx, ok := c.partitionDist[partID]
+	if !ok {
+		return nil
+	}
+	virtualNode := c.sortedRing[virtualNodeIdx]
+	mem, ok := c.nodeToMems[virtualNode]
+	assert.Assertf(ok, "member must exist for the virtual node")
+	return mem
 }
 
 func (c *ConsistentUniformHash) initializeVirtualNodes(members []Member) {
+	// Ensure the order of members to avoid inconsistency caused by hash collisions.
+	sort.Slice(members, func(i, j int) bool {
+		return members[i].String() < members[j].String()
+	})
+
 	for _, mem := range members {
 		for i := 0; i < c.config.ReplicationFactor; i++ {
 			// TODO: Shall use a more generic hasher which receives multiple slices or string?
-			key := []byte(fmt.Sprintf("%s%d", mem.String(), i))
+			key := []byte(fmt.Sprintf("%s%s%d", mem.String(), hashSeparator, i))
```

Comment:
Why use `hashSeparator`? Can it reduce the probability of hash collision?

Reply:
Make it hard to build the duplicate keys here.